concourse/scripts/spec/val_from_yaml_spec.rb

```diff
@@ -1,52 +1,73 @@
-require File.join(File.dirname(__FILE__), '..', 'val_from_yaml.rb')
+RSpec.describe "val_from_yaml.rb", :type => :aruba do
+  FIXTURE = File.expand_path("../fixtures/val_from_yaml.yml", __FILE__)
+
+  def run_with_fixture(arg)
+    run("./val_from_yaml.rb #{arg} #{FIXTURE}")
+  end
 
-RSpec.describe "PropertyTreeHelper" do
-  let(:test_property_tree) {
-    property_yaml = %q{
----
-foo:
- bar:
-  val1: a
-  val2: b
- array1:
- - name: item1
-   val: array1_item1_value
- - name: item2
-   val: array1_item2_value
- array2:
- - array2_value1
- - array2_value2
- - array2_value3
-}
-    PropertyTree.load_yaml(property_yaml)
-  }
   it "retrieves a simple value" do
-    expect(test_property_tree['foo.bar.val1']).to eq('a')
+    run_with_fixture("foo.bar.val1")
+    expect(last_command_started).to have_exit_status(0)
+    expect(last_command_started.output).to eq <<-EOF
+a
+    EOF
   end
+
   it "retrieves a full data structure" do
-    val = test_property_tree['foo.bar']
-    expect(val).to be_a(Hash)
-    expect(val).to include('val1', 'val2')
+    run_with_fixture("foo.bar")
+    expect(last_command_started).to have_exit_status(0)
+    expect(last_command_started.output).to eq <<-EOF
+---
+val1: a
+val2: b
+    EOF
   end
+
   it "returns nil for non existing properties" do
```

Comment:
I wonder whether this test description should be updated to reflect the commandline behaviour (exits non-zero with no output)